spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -776,9 +776,12 @@ def add_epochs(recording: se.RecordingExtractor, nwbfile=None,
         for epoch_name in recording.get_epoch_names():
             epoch = recording.get_epoch_info(epoch_name)
             if nwbfile.epochs is None:
+                end_frame = epoch['end_frame']
+                if epoch['end_frame'] == recording.get_num_frames():
+                    end_frame -= 1
                 nwbfile.add_epoch(
                     start_time=recording.frame_to_time(epoch['start_frame']),
-                    stop_time=recording.frame_to_time(epoch['end_frame']),
+                    stop_time=recording.frame_to_time(end_frame),
```

Comment:
How did we end up with an improper end frame to begin with? Perhaps that is the better place to fix this.

Reply:
I'll try to completely retrace it; this particular case was discovered via upstream `frame_to_time` conversions being routed through a `SubRecordingExtractor` of a `MultiRecordingTimeExtractor` (hence why we had epochs to begin with) containing the custom `SyntalosRecordingExtractor`. At all the steps, I didn't really notice an improper use from what I could see, maybe @alejoe91 knows more about that.

Reply:
Yeah I think that the problem here is that the frames within an epoch are `start_frame <= frames < end_frame`. So in order to get the right timestamps we should always use: `stop_time=recording.frame_to_time(epoch['end_frame'] - 1)`